src/praisonai-agents/praisonaiagents/llm/streaming_protocol.py

```
@@ -0,0 +1,387 @@
+    Google Gemini streaming adapter.
+    
+    Handles Gemini's streaming issues with tools and JSON parsing.
+    """
+    
+    def can_stream(self, *, tools: Optional[List[Dict[str, Any]]] = None, **kwargs) -> bool:
+        """Gemini has issues with streaming + tools."""
+        if tools:
+            return False  # Disable streaming when tools are present due to JSON parsing issues
+        return True
+    
+    def create_stream_unavailable_event(self, reason: str = None, **metadata) -> StreamEvent:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)